### PR TITLE
AP_Mount: Initialize backends after all mounts were added

### PR DESCRIPTION
**Pull request by olliw42**
setups with two mounts are not in the limelight, but ArduPilot supports that.

This change initializes the mounts only after they all were created, which is needed so they realize what the state of affairs is.

### CODE REVIEW

File: libraries/AP_Mount/AP_Mount.cpp [hunk 1]
```diff
@@ -470,13 +470,17 @@ void AP_Mount::init()
...
             if (!primary_set) {
                 _primary = instance;
                 primary_set = true;
             }
         }
     }
+
+    // init each instance, do it after all instances were created, so that they all know things
+    for (uint8_t instance=0; instance<AP_MOUNT_MAX_INSTANCES; instance++) {
+        if (_backends[instance] != nullptr) _backends[instance]->init();
```
Comment (khancyr) [inline libraries/AP_Mount/AP_Mount.cpp:482]:
```suggestion
        if (_backends[instance] != nullptr) {
            _backends[instance]->init();
        }
```

**Review by khancyr — CHANGES_REQUESTED** [review]
Mostly good
We don't use one line if so we need to enclose with {}

We are quite pedantic on commit message, it should start with the library name, so : 
AP_Mount: initialize backends after all mounts were added
You can use the script Tools/gittools/git-subsystem-split to append it for you !

File: libraries/AP_Mount/AP_Mount.cpp (hunk 1 above)
Comment (olliw42) [reply libraries/AP_Mount/AP_Mount.cpp]: fair enough. done.

**Comment by peterbarker** [comment]
@olliw42 seems a reasonable patch.

What problems were occurring that this fixes?

**Comment by olliw42** [comment]
none of the current backends would work as they are right now, so in this sense it doesn't solve anything

when you want to modify a backend so it can handle a dual mounts, or add one, you likely need this, e.g. if you want some level of self configuration. The practical example would be two mavlink speaking gimbals, which are identified by id, and do not in addition need to be identified by serial port.

btw, I do have this code snippet in use for nearly a year now, so it's not coming out of a sudden gut feeling... I just thought, hey, why not
it is such a simple change that I couldn't think about a reason why one would object it, so I PRed it
if you don't like it, no problem, I certainly can live with it ;)

**Comment by olliw42** [comment]
superseded by https://github.com/ArduPilot/ardupilot/pull/18440
hence closing

**Pull request closed without merging**